code/modules/mob/living/emote.dm

```diff
@@ -132,6 +132,17 @@
 				wings.open_wings()
 			addtimer(CALLBACK(wings, open ? /obj/item/organ/external/wings/functional.proc/open_wings : /obj/item/organ/external/wings/functional.proc/close_wings), wing_time)
 
+/datum/emote/living/flap/can_run_emote(mob/user, status_check = TRUE , intentional)
+	if(!..())
+		return FALSE
+	if (istype(user, /mob/living/simple_animal/chicken))
+		return TRUE
+	if(!istype(user, /mob/living/carbon/human))
+		return FALSE
```

Comment:
I'm not fond of these hardcoded checks: There are quite a few simple mobs that have wings and not just chicken (penguins, bees, drakes, bats, butterflies, ice whelps, chicks and watchers too come to my mind) and if we were to include all or most of them here, this would turn into an awful cascade of `if` conditional statement.
This is a suggestion, but there should be only one conditional statement on whether the mob has a trait that lets them use this emote. You would then need to make sure the trait is added to the simple mobs on `Initialize()` and that wings also give and remove this trait to/from the owner on `Insert()` and `Remove()` but that's it.

Reply:
Either way, emotes have a `mob_type_allowed_typecache`, `mob_type_blacklist_typecache`, and a `mob_type_ignore_stat_typecache` on `/emote`, which can be used instead